src/testing/flow_solver.cpp

```diff
@@ -1,35 +1,324 @@
+                    std::string::size_type found = line.find(name);
+                    
+                    // found the line corresponding to the desired parameter
+                    if (found!=std::string::npos) {
+
+                        // construct the updated line
+                        std::string updated_line = line;
+                        std::string::size_type position_to_replace = line.find_last_of("=")+2;
+                        std::string part_of_line_to_replace = line.substr(position_to_replace);
+                        updated_line.replace(position_to_replace,part_of_line_to_replace.length(),value_string);
+
+                        // write updated line to restart file
+                        RESTART_FILE << updated_line << "\n";
+                        
+                        // update the parameter index, name, and value
+                        if ((i_parameter+1) < number_of_subsection_parameters[i_subsection]) ++i_parameter; // to avoid going out of bounds
+                        if (i_subsection==0) {
+                            name = ODE_solver_restart_parameter_names[i_parameter];
+                            value_string = ODE_solver_restart_parameter_values[i_parameter];
+                        } else if (i_subsection==1) {
+                            name = flow_solver_restart_parameter_names[i_parameter];
+                            value_string = flow_solver_restart_parameter_values[i_parameter];
+                        }
+                    } else {
+                        // write line (that does correspond to the desired parameter) to the restart file
+                        RESTART_FILE << line << "\n";
+                    }
+                }
+                // update the subsection index
+                if ((i_subsection+1) < number_of_subsections) ++i_subsection; // to avoid going out of bounds
+            } else {
+                // write line (that is not in a desired subsection) to the restart file
+                RESTART_FILE << line << "\n";
+            }
+        }
+    }
+}
+
+#if PHILIP_DIM>1
+template <int dim, int nstate>
+void FlowSolver<dim,nstate>::output_restart_files(
+    const int current_restart_index,
+    const double constant_time_step,
+    const std::shared_ptr <dealii::TableHandler> unsteady_data_table) const
+{
+    pcout << "  ... Writing restart files ... " << std::endl;
+    const std::string restart_filename_without_extension = get_restart_filename_without_extension(current_restart_index);
+
+    // solution files
+    dealii::parallel::distributed::SolutionTransfer<dim, dealii::LinearAlgebra::distributed::Vector<double>, dealii::DoFHandler<dim>> solution_transfer(dg->dof_handler);
+    solution_transfer.prepare_for_coarsening_and_refinement(dg->solution);
+    dg->triangulation->save(flow_solver_param.restart_files_directory_name + std::string("/") + restart_filename_without_extension);
```

Comment:
I'm looking at
https://www.dealii.org/developer/doxygen/deal.II/classparallel_1_1distributed_1_1SolutionTransfer.html
subsection Use for Serialization

Is prepare_for_serialization() needed?

Reply:
I think so, I've changed it and the tests are still passing. `prepare_for_coarsening_and_refinement` was used in other parts of the code so that's why I had used it in the first place. Take a look at subsection "Note on usage with [DoFHandler](https://www.dealii.org/current/doxygen/deal.II/classDoFHandler.html) with hp-capabilities". It looks like that's what were actually doing since we're passing in that 3rd template variable; I'm passing it because it was done like that in other parts of the code; see line 30 of `src/grid_refinement/grid_refinement_uniform.cpp`

Reply:
Funnily enough the deal.II internally calls prepare_for_coarsening_and_refinement, however it packages it in a std::vector of vector in case you want to serialize multiple solutions.
https://github.com/dougshidong/dealii/blob/6abe6d7889e33a788aa30b29d98e7c6b205618c2/source/distributed/solution_transfer.cc#L180-L197

However, I think it makes sense to use prepare_for_serialization() instead because the function name is more relevant.